### PR TITLE
feat: ✨ DX: Angular should allow using boolean attributes without square brackets in templates

### LINKED ISSUE #689: feat: ✨ DX: Angular should allow using boolean attributes without square brackets in templates
# Checklist (for the issue creator)

- [x] I have looked at the documentation site and was not able to find the required feature.
- [x] I have looked into the Figma boards and was not able to find the required feature.
- [x] I would be willing to provide a PR for this feature.

# Story

As a Angular developer using the Synergy Design System, I would like to be able to use `boolean` attributes directly so they are easier to type.

**Before**:

```html
<syn-radio-group
  name="gender"
  label="Please tell us your gender"
  [required]="true"
>
  <!-- List of options -->
</syn-radio-group>
```

**After**:

```html
<syn-radio-group
  name="gender"
  label="Please tell us your gender"
  required
>
  <!-- List of options -->
</syn-radio-group>
```

This would be possible by using [Angular Input Coercion](https://angular.dev/tools/cli/template-typecheck#input-setter-coercion), e.g. allowing an empty string as a value for booleans.

Steps to do:

1. Provide a utility function in build that checks if an attribute is a `boolean` (e.g. `attribute.type.text === 'boolean'`
2. Use this utility function in angular builds to turn the old direct setter into a coerced method as detailed in the Angular Documentation link.

# Acceptance criteria

- [ ] All DoD items successfully completed
- [ ] Build runs through
- [ ] The "new" syntax is available
- [ ] The "old" square bracket syntax still works

### PR DESCRIPTION
**Pull request by schilchSICKAG**
# Pull Request

## 📖 Description

This PR provides a small change for our angular wrappers, adding the possibility to use short hand notation for `boolean` attributes without angulars... angular syntax :).

### Before:

```html
<syn-input name="something" [required]="true" />
<syn-checkbox name="checkbox" [required]="true" [checked]="true" />
```

### After:

```html
<syn-input name="something" required />
<syn-checkbox name="checkbox" required checked />
```

Both syntaxes are valid from now on, making it possible to use the shorter syntax in the future without breaking current projects relying on the angular-bracket notation.

### 🎫 Issues

Closes #689

## 👩‍💻 Reviewer Notes

Just have a look at the created code for the angular components. Also, play around with the angular demo project to see how this can be mixed and matched to taste.

## 📑 Test Plan

- Check out the repository
- Install and build all
- Move to the angular demo and start it up. This version already uses the short hand syntax where possible
- Change one of the required fields in a demo to the old syntax. Note that there is no difference between both versions.

## ✅ DoD

- [x] I have tested my changes manually.
- [x] I have added automatic tests for my changes (unit- and visual regression tests).
- [x] I have updated the project documentation to reflect my changes (e.g. CHANGELOG, Storybook, ...).
- [ ] ~~I have added documentation to the DaVinci migration guide (if need be)~~
- [x] I have made sure to follow the projects coding and contribution guides.
- [x] I have made sure that the bundle size has changed appropriately.
- [x] I have validated that there are no accessibility errors.
- [ ] ~~I have used design tokens instead of fix css values~~

### CODE REVIEW

File: packages/angular/README.md
Comment (kirchsuSICKAG) [inline packages/angular/README.md:1]: In line 256 is still the old way of using a boolean: 

```js
  [required]="true"
```

should be 

```js
  required
```

**Comment by kirchsuSICKAG** [comment]
In the file `angular-demo/src/app.component.html` is still the old syntax used (at line 16, 21 and 25)

**Comment by schilchSICKAG** [comment]
> In the file angular-demo/src/app.component.html is still the old syntax used (at line 16, 21 and 25)

See, it still works :). Just removed it, thank you for the catch

**Review by kirchsuSICKAG — CHANGES_REQUESTED** [review]

File: packages/angular/README.md
Comment (schilchSICKAG) [reply packages/angular/README.md:1]: Done

**Review by kirchsuSICKAG — APPROVED** [review]

**Pull request merged**